examples/deadline_writer.rs

```diff
@@ -0,0 +1,212 @@
+    buf.parse::<usize>()
+}
+
+fn main() {
+    let handle = LocalExecutorBuilder::new().pin_to_cpu(0)
+        .spawn(|| async move {
+
+            let cpuhog_tq = Local::create_task_queue(Shares::Static(1000), Latency::NotImportant, "cpuhog");
+
+            println!("{}", Style::new().bold().paint("Welcome to the Deadline Writer example"));
+            println!("In this example we will write a sequence of integers to a variable, busy looping for 500us after each write");
+            println!("While we do that, another CPU hog will be running constantly in a different TaskQueue");
+            println!("For {} results, this test is pinned to your CPU0. Make sure nothing else of significance is running there. You should be able to see it at 100% at all times!",
+                     Style::new().bold().paint("best"));
+
+            println!("\n\nPlease tell me how many integers you would like to write");
+            let to_write = blocking_read_int().unwrap();
+            println!("Ok, now let's write {} integers with both the writer and the CPU hog having the same priority", Colour::Blue.paint(to_write.to_string()));
+            let dur = static_writer(to_write, 1000, cpuhog_tq).await;
+            println!("Finished writing in {}", Colour::Green.paint(format!("{:#.0?}", dur)));
+            println!("This was using {} shares, and short of reducing the priority of the CPU hog. {}",
+                Colour::Green.paint("1000"), Style::new().bold().paint("This is as fast as we can do!"));
+            println!("With {} shares, this would have taken approximately {}", Colour::Green.paint("100"), Colour::Green.paint(format!("{:#.1?}", dur * 10)));
+            println!("With {} shares, this would have taken approximately {}. {}.", Colour::Green.paint("1"),
+                     Colour::Green.paint(format!("{:#.1?}", dur * 1000)), Style::new().bold().paint("Can't go any slower than that!"));
+
+            println!("\n\nLet's try the controlled process. How long would you like it to take? (seconds)");
+            println!("Keep in mind that very short processes will be inherently unstable because of the time the controller needs to adapt");
+            let mut duration = blocking_read_int().unwrap();
+            loop {
+
+                let deadline = Rc::new(DeadlineQueue::new("example", Duration::from_millis(250)));
+                let writer = Task::local(enclose!{ (deadline) async move {
+                    let test = IntWriter::new(to_write, Duration::from_secs(duration as u64));
+                    deadline.push_work(test).await.unwrap()
+                }}).detach();
+
+                Local::later().await;
+
+                let stop = Rc::new(Cell::new(false));
+                let hog = competing_cpu_hog(stop.clone(), cpuhog_tq);
+                let dur = writer.await.unwrap();
+                println!("Finished writing in {}", Colour::Green.paint(format!("{:#.2?}", dur)));
+                stop.replace(true);
```

Comment:
Is there any particular reason to use `.replace` over `.set`? This feels slightly unusual.

Reply:
no, just force of habit as sometimes I do want the value back then end up always using replace from muscle memory. Will change.